src/main/java/com/ridemart/service/UserService.java

```diff
@@ -0,0 +1,75 @@
+package com.ridemart.service;
+
+import com.ridemart.entity.User;
+import com.ridemart.exception.DuplicateUserException;
+import com.ridemart.exception.InvalidCredentialsException;
+import com.ridemart.exception.UserNotFoundException;
+import com.ridemart.repository.UserRepository;
+import com.ridemart.util.PasswordUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
+
+    @Autowired
```

Comment:
You need to follow a unified approach for dependency injection
in your other classes, you inject dependencies through the constructor
please, use this approach here as well